File: 445-add-two-numbers-ii/add-two-numbers-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        Stack<Integer> stack1 = new Stack<Integer>();
        ListNode p = l1;
        while (p!=null){
            stack1.push(p.val);
            p = p.next;
        }
        Stack<Integer> stack2 = new Stack<Integer>();
        
        p = l2;
        while (p!=null){
            stack2.push(p.val);
            p = p.next;
        }

        int carry = 0;

        ListNode prev = null;
        while (stack1.size() != 0 || stack2.size() != 0){
            int val = 0;
            if(stack1.size() != 0){
                val += stack1.pop();
            }
            if(stack2.size() != 0){
                val += stack2.pop();
            }

            val += carry;
            carry = val /10;
            val %= 10;

            ListNode n = new ListNode(val);
            n.next = prev;
            prev = n;
        }
        
        if(carry ==1){
            ListNode n = new ListNode(carry);
            n.next = prev;
            prev = n;
        }

        return prev;
    }
}

File: 445-add-two-numbers-ii/add-two-numbers-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        stack1 = []
        p = l1
        while p:
            stack1.append(p.val)
            p = p.next
        
        stack2 = []
        p = l2
        while p:
            stack2.append(p.val)
            p = p.next
        
        prev = None
        carry = 0
        while stack1 or stack2:
            val = 0
            if stack1:
                val += stack1.pop()
            if stack2:
                val += stack2.pop()
            
            val += carry
            carry = val // 10
            val = val % 10

            n = ListNode(val)
            n.next = prev
            prev = n
        

        if carry:
            n = ListNode(carry)
            n.next = prev
            prev = n
        return prev